lib/src/Uncertainty/Algorithm/Sensitivity/JansenSensitivityAlgorithm.cxx

```diff
@@ -115,6 +115,73 @@ Sample JansenSensitivityAlgorithm::computeIndices(const Sample & sample,
   return varianceI;
 }
 
+void JansenSensitivityAlgorithm::computeAsymptoticDistribution() const
+{
+  const UnsignedInteger inputDimension = inputDesign_.getDimension();
+  const UnsignedInteger outputDimension = outputDesign_.getDimension();
+
+  // psi
+  Description X(Description::BuildDefault(outputDimension, "X"));
+  Description Y(Description::BuildDefault(outputDimension, "Y"));
+  Description XY(2 * outputDimension);
+  String sumX;
+  String sumY;
+  for (UnsignedInteger q = 0; q < outputDimension; ++ q)
+  {
+    XY[2 * q] = X[q];
+    XY[2 * q + 1] = Y[q];
+    sumX += X[q];
+    sumY += Y[q];
+    if (q < outputDimension - 1)
+    {
+      sumX += "+";
+      sumY += "+";
+    }
+  }
+  sumX = "(" + sumX + ")";
+  sumY = "(" + sumY + ")";
+  Function psiFO = SymbolicFunction(XY, Description(1, "1 - 0.5*" + sumX + "/" + sumY));
+  Function psiTO = SymbolicFunction(XY, Description(1, "0.5*" + sumX + "/" + sumY));
+
+  Point varianceFO(inputDimension);
+  Point varianceTO(inputDimension);
+
+  for (UnsignedInteger p = 0; p < inputDimension; ++ p)
+  {
+    Sample uFO(size_, 0);
+    Sample uTO(size_, 0);
```

Comment:
For performance reasons, you should avoid Sample::stack.  Allocate uFO and uTO only once outside p-loop, and fill these Sample in these loops; yemB and yAmB are then no more needed.

Reply:
If this makes code much less readable, an alternative is to work on transposed Sample, and replace  Sample::stack by Sample::add.

Reply:
I prefer the code to be maintainable and readable here. Not sure if option 2 would be very readable too, besides one still have to transpose to compute the covariance.

Reply:
I started to implement your solution, and it looks good. I can even swap the i and q loops to follow the sample memory layout.

Reply:
Thanks